### DISCUSSION TITLE
Python Code in Github Pages - Noob needs help

### DISCUSSION THREAD

**Discussion by spulick** [community/community#21596] [Other Feature Feedback, Questions, & Ideas]
<p>Hello there,</p>
<p>I am making a webpage for a report of a study. I would like people to be able to view plots based on inputs they give. I have that code ready in Python. How do I add it to my Github Page?</p>
<p>Thanks in advance.</p>

**Comment by airtower-luna** [comment]
<p>That is not possible. GitHub Pages support only static pages, nothing that runs code on the server. For that you’ll have to find a host that lets you deploy web applications, or containers, or however else you’d like to deploy your service.</p>

**Comment by spulick** [comment]
<p>Thank you for your reply.</p>
<p>I went on GitHub Pages because I watched this video. <a href="https://youtu.be/2MsN8gpT6jY?t=38" rel="noopener nofollow ugc">https://youtu.be/2MsN8gpT6jY?t=38</a>. This is what my output might look like. How do you think these guys did it?</p>
<p>thanks again.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>That diagram is probably animated with CSS and Javascript, so the animation is running on the client side. If you can do it all in Javascript, a user could provide data to the script running in their browser and have it rendered, but it’d be gone as soon as they close the page (though you could include some export).</p>
<p>Or you could store the data to generate diagrams in your repository, and use <a href="https://docs.github.com/en/actions">Actions</a> to build a static site using your Python code and commit that to the branch published by Pages. In that case users could suggest datasets to add via pull requests.</p>

**Comment by spulick** [comment]
<p><a class="mention" href="/airtower-luna">@airtower-luna</a> Thank you so much for your help.</p>

**Comment by Apostolis-Triantafyllou** [comment]
https://pyscript.net/ is an option
